trieste/acquisition/function.py

```diff
@@ -509,6 +512,155 @@ def prepare_acquisition_function(
         return lambda at: expected_improvement(objective_model, eta)(at) * constraint_fn(at)
 
 
+class ExpectedHypervolumeImprovement(SingleModelAcquisitionBuilder):
+    """
+    Builder for the :func:`expected_hv_improvement` acquisition function.
```

Comment:
I think I have misunderstood something here. But are we not in the multi-model case when doing Multi-objective BO? Perhaps @vpicheny can explain?

Reply:
you can either do multiobj with a single multi-output model or a set of single-output models (or both...).

Reply:
I would also expect a multi-model by default

Reply:
I don't think we support the set of single-output models case at the moment though, as we only allow a single objective function tag.

This isn't necessarily an issue. We can stack single output models using the model stacker to get a single model for all the objectives. 

Perhaps we leave the multi-model case for constraints? I think this is what Joel wanted.

Reply:
the reason is that EHVI assumes all outputs used in EHVI can be combined in the same tensor. Thus the corresponding data can be combined in the same dataset. It seemed like an unnecessary freedom involving too much boilerplate and other complexity to allow multiple datasets

Reply:
Thanks @joelberkeley 

I agree completely. So @TsingQAQ no changes required here.

Reply:
sounds good!